openmc/mgxs/mgxs.py

```diff
@@ -0,0 +1,5909 @@
+from collections import OrderedDict
+from numbers import Integral
+import warnings
+import os
+import copy
+from abc import ABCMeta
+
+import numpy as np
+import h5py
+
+import openmc
+import openmc.checkvalue as cv
+from openmc.tallies import ESTIMATOR_TYPES
+from openmc.mgxs import EnergyGroups
+
+
+# Supported cross section types
+MGXS_TYPES = ['total',
+              'transport',
+              'nu-transport',
+              'absorption',
+              'capture',
+              'fission',
+              'nu-fission',
+              'kappa-fission',
+              'scatter',
+              'nu-scatter',
+              'scatter matrix',
+              'nu-scatter matrix',
+              'multiplicity matrix',
+              'nu-fission matrix',
+              'scatter probability matrix',
+              'consistent scatter matrix',
+              'consistent nu-scatter matrix',
+              'chi',
+              'chi-prompt',
+              'inverse-velocity',
+              'prompt-nu-fission',
+              'prompt-nu-fission matrix']
+
+# Supported domain types
+DOMAIN_TYPES = ['cell',
+                'distribcell',
+                'universe',
+                'material',
+                'mesh']
+
+# Filter types corresponding to each domain
+_DOMAIN_TO_FILTER = {'cell': openmc.CellFilter,
+                     'distribcell': openmc.DistribcellFilter,
+                     'universe': openmc.UniverseFilter,
+                     'material': openmc.MaterialFilter,
+                     'mesh': openmc.MeshFilter}
+
+# Supported domain classes
+_DOMAINS = (openmc.Cell,
+            openmc.Universe,
+            openmc.Material,
+            openmc.RegularMesh)
+
+# Supported ScatterMatrixXS angular distribution types
+MU_TREATMENTS = ('legendre', 'histogram')
```

Comment:
The string `legendre` is used ~108 times in this. I think you should make it a constant.